pkg/skaffold/build/local/prune.go

```diff
@@ -160,7 +166,11 @@ func (p *pruner) collectImagesToPrune(ctx context.Context, artifacts []*latest.A
 		if err != nil {
 			switch err {
 			case context.Canceled, context.DeadlineExceeded:
-				return nil
+				// ctx is done or cancelled or timed out
+				// Usually means the process is interrupted
+				// Skip logging and return immediately
+				// https://github.com/GoogleContainerTools/skaffold/issues/4888
```

Comment:
I'm normally all for adding comments, but this is self-evident given the err values.  I'd suggest removing it.

Reply:
Removed the comment.

It could be a bit unclear how we ended up with a cancelled context. Having a comment about Ctl-C pressed could help to understand why we're here and what's the best way to handle the situation. But, on the other hand, checks like this are pretty common in our code, and there's no need to comment each one.

Reply:
There are a number of ways that a context may be cancelled — we may even cancel it programmatically (e.g., file change and so we’re going to trigger another cycle of the dev loop). Such is life :-)